src/softmax.jl

```diff
@@ -142,3 +142,9 @@ function logsumexp(x::AbstractArray; dims = :)
     max_ = maximum(x; dims = dims)
     max_ .+ log.(sum(exp.(x .- max_); dims = dims))
 end
+
+# Informative error message if any of the softmax variants is called with a number
+for f in (:softmax, :logsoftmax, :softmax!, :logsoftmax!, :logsumexp)
+    @eval $(f)(x::Number, args...) = 
+      error("(`", $(string(f)), "(x)`) called with a number, but it expects an array.")
```

Comment:
Maybe one less set of brackets? And slightly more explanation, since the user presumably did not intend to apply this to one number, so may not know what to change. Maybe something like this:
```suggestion
      error("`", $(string(f)), "(x)` called with a number, but it expects an array. Usually this is because a layer like `Dense(3,4,softmax)` is broadcasting it like an activation function; `softmax` needs to be outside the layer.")
```